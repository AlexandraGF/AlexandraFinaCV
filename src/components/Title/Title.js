import React, { Component } from 'react';
import './Title.scss';

class Title extends Component {
  render() {
      const linkedIn = "https://www.linkedin.com/in/alexandra-fina-0418aab2/";
      const gitHub = "https://github.com/AlexandraGF";
    return (
      <div className="title">
        <div>Alexandra Fina</div>
        <div className="anchors">
            <a href={linkedIn} target="_blank"><div className="linkedInAnchor">LinkedIn</div></a>
            <a href={gitHub} target="_blank"><div className="gitHubAnchor">GitHub</div></a>
        </div>
      </div>
    );
  }
}

export default Title;
