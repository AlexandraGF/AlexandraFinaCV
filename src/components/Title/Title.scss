@import '../../variables.scss';

.title {
    font-family: 'Allura', cursive;
    font-size: 30px;
    margin: 6% 4% 6% 0%;
    padding-left: 6%;
    text-decoration: underline;
}

.anchors {
    font-family: 'Quintessential', cursive;
    font-size: 15px;
    display: flex;
    flex-wrap: wrap;
    margin-top: 4%;
    margin-left: 20%;
}

.anchors > * {
    flex: 0 0 38%;
}

.linkedInAnchor:hover, .gitHubAnchor:hover {
    transform: scale(1.4);
    cursor: $cursorPointer;
}

.linkedInAnchor, .gitHubAnchor {
    color: black;
    list-style: none;
    display: inline-block;
    padding: 0px 20px;
    width: 45%;
    box-shadow: 25px 10px 12px $boxShadowColor;
    border-radius: $borderRadius;
}
