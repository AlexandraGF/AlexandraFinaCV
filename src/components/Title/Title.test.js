import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

import Title from './Title';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Title />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders text', () => {
    const wrapper = mount(<Title />);
    expect(wrapper.text()).toContain("Alexandra");
    });
    
    it('renders Title Component', () => {
    const tree = renderer.create(
    <Title />
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
    })
