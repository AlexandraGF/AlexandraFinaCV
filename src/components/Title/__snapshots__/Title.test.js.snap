// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App Component 1`] = `
<div
  className="title"
>
  <div>
    Alexandra Fina
  </div>
  <div
    className="anchors"
  >
    <a
      href="https://www.linkedin.com/in/alexandra-fina-0418aab2/"
      target="_blank"
    >
      <div
        className="linkedInAnchor"
      >
        LinkedIn
      </div>
    </a>
    <a
      href="https://github.com/AlexandraGF"
      target="_blank"
    >
      <div
        className="gitHubAnchor"
      >
        GitHub
      </div>
    </a>
  </div>
</div>
`;

exports[`renders Title Component 1`] = `
<div
  className="title"
>
  <div>
    Alexandra Fina
  </div>
  <div
    className="anchors"
  >
    <a
      href="https://www.linkedin.com/in/alexandra-fina-0418aab2/"
      target="_blank"
    >
      <div
        className="linkedInAnchor"
      >
        LinkedIn
      </div>
    </a>
    <a
      href="https://github.com/AlexandraGF"
      target="_blank"
    >
      <div
        className="gitHubAnchor"
      >
        GitHub
      </div>
    </a>
  </div>
</div>
`;
